### DISCUSSION TITLE
Feature Request: A Knowledge Base or FAQ from comments + code snippets in Pull Requests

### DISCUSSION THREAD

**Discussion by svsool** [community/community#70625] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Often in reviews there are various discussions brought up about do's and don'ts, and with newcomers and people onboarded those things keep re-surfacing over and over again.

A manual approach of collecting comments and converting it to a contribution guideline serves as a good start.

Idea: Add a possibility to search through comments + mark comments as favourite ones + tagging, so it can serve as a knowledge base or FAQ later to new people who are contributing to the project. Great if both code before and after can be referenced, annotating manually fine as well if code changed too much or was moved.

Having a code snippet along with the comment is great for context and understanding why certain practice should be used by (new) contributors.

Better if such knowledge base could be automatically organized from existing comments using some LLM model or "ask how to do X based on previous PR comments / code" with the reference where such issue was discussed.

**Comment by queenofcorgis** [comment]
Welcome to the GitHub Community, @svsool , we're happy you're here! You are more likely to get a useful response if you are posting your feedback in the applicable category. I've gone ahead and moved it for you. Good luck!
